nuget/OpenTelemetry.AutoInstrumentation.Dependencies/OpenTelemetry.AutoInstrumentation.Dependencies.nuspec

```diff
@@ -18,6 +18,7 @@
     <dependencies>
       <!-- TODO: Need a test/validation to make sure that this are in sync with actual projects -->
       <group targetFramework="net6.0">
+        <dependency id="MongoDB.Driver.Core.Extensions.DiagnosticSources" version="1.3.0" />
```

Comment:
The package should be added to the `OpenTelemetry.AutoInstrumentation` project instead. There is some overlap with the dependencies package, and we may end up dropping one of them. Anyway, for now the right place for this is on `OpenTelemetry.AutoInstrumentation`.

Reply:
Adding it to `OpenTelemetry.AutoInstrumentation` will make the dependencies shipping even worse. Nulling all optimizations like https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2294

Reply:
That is true @RassK I think the path to addressing that is via #2329 - right now the NuGet package is broken for MongoDB.

Reply:
I am checking second option locally:
 - move MogoDB.Instrumentation as a dependency from Additional Store directly to instrumentation (as side effect .NET Fx should work)
 - OpenTelemetry.AutoInstrumentation will be referencing MongoDb.Instrumentation.

Reply:
> @RassK, we don't need to add it to `OpenTelemetry.AutoInstrumentation` __csproj__ we can temporarily have it directly on the `OpenTelemetry.AutoInstrumentation` __nuspec__. 

I think this is the only reasonable idea for now. This is one of the worst direct references to take for our project.

Reply:
See discussion at https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/2397#issuecomment-1505423109